pandas/core/reshape/merge.py

```diff
@@ -1089,13 +1089,19 @@ def _validate_specification(self):
                     raise ValueError('len(left_on) must equal the number '
                                      'of levels in the index of "right"')
                 self.right_on = [None] * n
+            if not self.right_on:
```

Comment:
can we instead try to set

self.left_on = self.left_on or  [None] * len( self.right_on)
and same for right_on

before these if clauses; that way don’t need to add additional checks

Reply:
Sure